client/src/main/java/io/split/client/HttpSplitChangeFetcher.java

```diff
@@ -59,27 +73,68 @@ long makeRandomTill() {
     @Override
     public SplitChange fetch(long since, long sinceRBS, FetchOptions options) {
         long start = System.currentTimeMillis();
+        SplitHttpResponse response;
         try {
+            if (SPEC_VERSION.equals(SPEC_1_1) && (System.currentTimeMillis() - _lastProxyCheckTimestamp >= PROXY_CHECK_INTERVAL_MILLISECONDS_SS)) {
+                _log.info("Switching to new Feature flag spec ({}) and fetching.", SPEC_1_3);
+                SPEC_VERSION = SPEC_1_3;
+            }
             URI uri = buildURL(options, since, sinceRBS);
-            SplitHttpResponse response = _client.get(uri, options, null);
-
+            response = _client.get(uri, options, null);
             if (response.statusCode() < HttpStatus.SC_OK || response.statusCode() >= HttpStatus.SC_MULTIPLE_CHOICES) {
                 if (response.statusCode() == HttpStatus.SC_REQUEST_URI_TOO_LONG) {
                     _log.error("The amount of flag sets provided are big causing uri length error.");
                     throw new UriTooLongException(String.format("Status code: %s. Message: %s", response.statusCode(), response.statusMessage()));
                 }
 
+                if (response.statusCode() == HttpStatus.SC_BAD_REQUEST && SPEC_VERSION.equals(Spec.SPEC_1_3) && _rootURIOverriden) {
+                    SPEC_VERSION = Spec.SPEC_1_1;
+                    _log.warn("Detected proxy without support for Feature flags spec {} version, will switch to spec version {}",
+                            SPEC_1_3, SPEC_1_1);
+                    _lastProxyCheckTimestamp = System.currentTimeMillis();
+                    return fetch(since, sinceRBS, options);
+                }
+
                 _telemetryRuntimeProducer.recordSyncError(ResourceEnum.SPLIT_SYNC, response.statusCode());
                 throw new IllegalStateException(
                         String.format("Could not retrieve splitChanges since %s; http return code %s", since, response.statusCode())
                 );
             }
-            return Json.fromJson(response.body(), SplitChange.class);
         } catch (Exception e) {
             throw new IllegalStateException(String.format("Problem fetching splitChanges since %s: %s", since, e), e);
         } finally {
             _telemetryRuntimeProducer.recordSyncLatency(HTTPLatenciesEnum.SPLITS, System.currentTimeMillis() - start);
         }
+
+        SplitChange splitChange = new SplitChange();
+        if (SPEC_VERSION.equals(Spec.SPEC_1_1)) {
+            splitChange.featureFlags = convertBodyToOldSpec(response.body());
+            splitChange.ruleBasedSegments = createEmptyDTO();
+        } else {
+            splitChange = Json.fromJson(response.body(), SplitChange.class);
+        }
+        return splitChange;
```

Comment:
why did you move this out of the try and catch ? I think that everything should go inside